### PR TITLE
feat: make cxx core more extensible

### PR DESCRIPTION
**Pull request by jpudysz**
## Summary

Make cxx core more extensible for Cxx TurboModules and RN 0.75

Min required versions:
- Expo 51 ✅
- react-native 0.74 ✅
- web (no changes) ✅
- SSR (no changes) ✅
- Android  ✅
- iOS ✅
- react-native-macos (awaiting 0.74 release, should work out of the box) ❓
- react-native-visionos (in testing) 🟡
- react-native-tvos (in testing) 🟡
- Windows (in progress) ❌

### CODE REVIEW

**Comment by jpudysz** [comment]
Min required versions:
- Expo 51 ✅
- react-native 0.74 ✅
- web (no changes) ✅
- SSR (no changes) ✅
- react-native-macos (awaiting 0.74 release, should work out of the box) ❓
- react-native-visionos (in testing) 🟡
- react-native-tvos (in testing) 🟡
- Android (in progress) ❌

**Comment by efstathiosntonas** [comment]
@jpudysz thank you for this: [android set root view background color](https://github.com/jpudysz/react-native-unistyles/pull/203/commits/0711d3150249aa5ac0c26926d1bc48158286112d)

**Comment by jpudysz** [comment]
No problem 😇

File: src/core/UnistylesRuntime.ts [hunk 1]
```diff
@@ -122,6 +149,41 @@ export class UnistylesRuntime {
...
+
+    /**
+     * Get the font scale
+     * @returns - The font scale
+     */
+    public get fontScale() {
+        const fontScale = this.unistylesBridge.fontScale
+
+        return fontScale === 1.0 ? 1 : fontScale.toFixed(2)
+    }
```
Comment (levibuzolic) [inline src/core/UnistylesRuntime.ts:168]: @jpudysz thanks so much for implementing this in 2.8.0 🙇‍♂️ 

Have just started playing with the new version and noticed `fontScale` was `number | string`, which prompted me to go look at the implementation.

Was the `toFixed` here to serve a specific purpose? Wondering if it'd make sense to always return a precise number and leave any rounding/clamping behaviour for userland implementations?

For my use of this value, I'd end up just wanting to `parseFloat()` this anyway.

File: src/core/UnistylesRuntime.ts [hunk 2]
```diff
@@ -1,14 +1,39 @@
...
  * Utility to interact with the Unistyles during runtime
  */
 export class UnistylesRuntime {
     constructor(private unistylesBridge: UnistylesBridge, private unistylesRegistry: UnistyleRegistry) {}
 
+    /**
+     * Get the mini runtime injected to creteStyleSheet
+     * @returns - The mini runtime
+     */
+    public get miniRuntime() {
```
Comment (levibuzolic) [inline src/core/UnistylesRuntime.ts:17]: Love this change by the way. 👏

File: src/core/UnistylesRuntime.ts (hunk 1 above)
Comment (jpudysz) [reply src/core/UnistylesRuntime.ts:168]: Oh no 😬

Native side reports float with 10 digits - that's why I though about rounding it. I didn't check the type and now it's string 🤦

I will fix it in today's release. 

Not sure if that amount of precision is required though

File: src/core/UnistylesRuntime.ts (hunk 2 above)
Comment (jpudysz) [reply src/core/UnistylesRuntime.ts:17]: Thanks!
Btw. Im always looking for your ideas like with font scale and pixel ratio

**Pull request merged**